WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift

```diff
@@ -127,6 +132,9 @@ final class DashboardViewModel: ObservableObject {
         setupDashboardCards()
         installPendingThemeIfNeeded()
         observeDashboardCardsAndReload()
+        Task {
+            await checkInboxEligibility()
+        }
```

Comment:
❓  Would adding this into `syncDashboardEssentialData` work? i.e. Check eligibility in viewDidLoad instead of view model's `init.

Reply:
I did this initially but there are complications when writing tests. We'd need to mock responses for `syncDashboardEssentialData`, including the requests for Blaze, which is not trivial as we discussed the other day. I'm keeping this implementation for now, we can always tidy up the code after deciding on the solution for testing Blaze.